### PR TITLE
#158487968 Implement App Tour for Players

### PR DESCRIPTION
**Pull request by iverenshaguy**
#### What does this PR do?
This pull request implements the gaming page tour for new players

#### Description of Task to be completed?
- implement app tour of gaming page
- change notifications to toaster

#### How should this be manually tested?
N/A

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
 #158487968

#### Screenshots (if appropriate)
screen shot 2018-07-17 at 5 26 56 pm

screen shot 2018-07-18 at 10 46 04 am

#### Questions:
N/A

### CODE REVIEW

File: client/js/controllers/game.js [hunk 1]
```diff
@@ -304,8 +303,6 @@ angular.module('mean.system')
 
           Global.getUser().then(function (response) {
             if (response.authenticated) {
-              console.log(response.authenticated);
-              console.log(donation);
```
Comment (otseobande) [inline client/js/controllers/game.js:308]: Good job removing the `console.log` statements.

File: client/views/app.html [hunk 2]
```diff
@@ -30,7 +30,7 @@
             <p>{{currentPlayer().username}}</p>
           </div>
         </li>
-        <li><a class="waves-effect waves-yellow btn tour-btn">Take A Tour</a></li>
+        <li><a class="waves-effect waves-yellow btn tour-btn" ng-click="CallMe()">Take A Tour</a></li>
```
Comment (otseobande) [inline client/views/app.html:33]: `callMe()` is definitely a funny function name 🙂. Is there any reason it's named that way?

File: client/views/question.html [hunk 3]
```diff
@@ -1,35 +1,38 @@
...
 >
   <div class="waiting-text">
+    <div id="searching-for-players-text">
     <p>Searching for Players</p>
     <div>
       <img src="/img/loaders/spinner.svg" width="50"/>
     </div>
-    <p class="text-center">{{game.players.length}}&nbsp;/&nbsp;12 Players Found</p>
+  </div>
+    <p class="text-center" id="players-found-text">{{game.players.length}}&nbsp;/&nbsp;12 Players Found</p>
```
Comment (otseobande) [inline client/views/question.html]: I think we could use other techniques to add whitespaces other than`&nbsp;`. We could use the `<pre>` tag for retaining the format in which we wrote the text in or use a `white-space` CSS property to achieve the same effect. `&nbsp;` has become one of the various HTML artifacts. I feel we should join the evolution of the web and use these other techniques.

File: client/views/tourview.html [hunk 4]
```diff
@@ -423,17 +359,9 @@
...
-          <p class="text-center">This game is based on the <span class="footer-text">Card Against Humanity</span> card game, which is CC licensed (BY-NC-SA). Cards For Humanity is not affiliated with Cards Against Humanity.</p>
-        </div>
-      </div>
-  <script src="https://code.jquery.com/jquery-3.3.1.min.js"></script>
-  <script src="https://cdnjs.cloudflare.com/ajax/libs/materialize/1.0.0-rc.2/js/materialize.min.js"></script>
-  <script src="../js/gamepage.js"></script>
-</body>
-</html>
\ No newline at end of file
+      </div>
```
Comment (otseobande) [inline client/views/tourview.html]: Just like Victoria would always note. We missed an extra line here.

**Review by otseobande — COMMENTED** [review]
You implemented this feature impressively. However, there are somethings that I still need to clarify. I noticed that the element attribute of an object in the steps array in the TourController uses an id selector. This makes sense as ids are a way to mark unique elements but this has led to us having ids and classes with the same names on the elements which the stepper is implemented on. Do you think its a good idea if we remove the class names and convert the styles to select the elements using these ids instead?

File: client/views/app.html (hunk 2 above)
Comment (iverenshaguy) [reply client/views/app.html:33]: It's not my function, it's from the angular-intro plugin. You can go to their plugin page to research why they decided to call their function CallMe(). https://github.com/mendhak/angular-intro.js

File: client/views/question.html (hunk 3 above)
Comment (iverenshaguy) [reply client/views/question.html]: Ok

**Comment by iverenshaguy** [comment]
I always use classes to style elements so that the modules are reusable. I used IDs for the angular-intro plugin because if I were to use classes, I'd need to append the position of each element as in ('.elem')[0] and I didn't want to get confused. I will change the id names to unique names so that they are not the same as the class names.

File: client/views/tourview.html (hunk 4 above)
Comment (iverenshaguy) [reply client/views/tourview.html]: We? I missed an extra line here. Fixing it now. Thanks

File: client/views/app.html (hunk 2 above)
Comment (otseobande) [reply client/views/app.html:33]: Alright

File: client/views/question.html (hunk 3 above)
Comment (iverenshaguy) [reply client/views/question.html]: `<pre>` changed it to a monospaced font

Comment (iverenshaguy) [reply client/views/question.html]: The `white-space` property also didn't work. However, I was able to remove the nbsp; and leave a space between and it maintained the space just like a normal sentence.

**Review by Stylll — APPROVED** [review]
It would have been nice if we had a little demo of the tour in form of a gif in your PR rather than just a screenshot. Nevertheless, you've done a good job. Keep it up.

**Review by otseobande — APPROVED** [review]

**Pull request merged**